### PR TITLE
BATM-2923 bitfinex unknown symbol - do not log full stacktrace

### PR DESCRIPTION
**Pull request by pvyhnal-generalbytes**

### CODE REVIEW

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitfinex/BitfinexExchange.java [hunk 1]
```diff
@@ -371,6 +374,8 @@ public String sellCoins(BigDecimal cryptoAmount, String cryptoCurrency, String f
             if (orderProcessed) {
                 return orderId;
             }
+        } catch (CurrencyPairNotValidException e) {
+            log.error("Currency pair not valid: {}, {}", currencyPair, e.getMessage());
```
Comment (generalbytes) [inline server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitfinex/BitfinexExchange.java:378]: Should be warning

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitfinex/BitfinexExchange.java [hunk 2]
```diff
@@ -155,10 +155,13 @@ public synchronized BigDecimal getExchangeRateLast(String cryptoCurrency, String
...
     private BigDecimal getExchangeRateLastSync(String cryptoCurrency, String cashCurrency) {
         MarketDataService marketDataService = getExchange().getMarketDataService();
+        CurrencyPair currencyPair = new CurrencyPair(getExchangeSpecificSymbol(cryptoCurrency), getExchangeSpecificSymbol(cashCurrency));
         try {
             RateLimiter.waitForPossibleCall(getClass());
-            Ticker ticker = marketDataService.getTicker(new CurrencyPair(getExchangeSpecificSymbol(cryptoCurrency), getExchangeSpecificSymbol(cashCurrency)));
+            Ticker ticker = marketDataService.getTicker(currencyPair);
             return ticker.getLast();
+        } catch (CurrencyPairNotValidException e) {
+            log.error("Currency pair not valid: {}, {}", currencyPair, e.getMessage());
```
Comment (generalbytes) [inline server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitfinex/BitfinexExchange.java:164]: Should be warning

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitfinex/BitfinexExchange.java [hunk 3]
```diff
@@ -281,6 +284,8 @@ public String purchaseCoins(BigDecimal amount, String cryptoCurrency, String fia
             if (orderProcessed) {
                 return orderId;
             }
+        } catch (CurrencyPairNotValidException e) {
+            log.error("Currency pair not valid: {}, {}", currencyPair, e.getMessage());
```
Comment (generalbytes) [inline server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitfinex/BitfinexExchange.java:288]: Should be warning

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitfinex/BitfinexExchange.java (hunk 1 above)
Comment (generalbytes) [reply server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitfinex/BitfinexExchange.java:378]: Should be warning

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitfinex/BitfinexExchange.java [hunk 4]
```diff
@@ -426,6 +430,8 @@ public boolean onCreate() {
                 } catch (InterruptedException e) {
                     log.error("Error", e);
                 }
+            } catch (CurrencyPairNotValidException e) {
+                log.error("Currency pair not valid: {}, {}", currencyPair, e.getMessage());
```
Comment (generalbytes) [inline server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitfinex/BitfinexExchange.java:434]: Should be warning

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitfinex/BitfinexExchange.java [hunk 5]
```diff
@@ -549,6 +554,8 @@ public boolean onCreate() {
...
```
Comment (generalbytes) [inline server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitfinex/BitfinexExchange.java:558]: Should be warning

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitfinex/BitfinexExchange.java [hunk 6]
```diff
@@ -693,10 +700,8 @@ public int compare(LimitOrder lhs, LimitOrder rhs) {
                 log.debug("Called Bitfinex exchange for BUY rate: " + cryptoCurrency + fiatCurrency + " = " + tradableLimit);
                 return tradableLimit.multiply(cryptoAmount);
             }
-        } catch (ExchangeException e) {
-            log.error("Error", e);
-        } catch (IOException e) {
-            log.error("Error", e);
+        } catch (CurrencyPairNotValidException e) {
+            log.error("Currency pair not valid: {}, {}", currencyPair, e.getMessage());
```
Comment (generalbytes) [inline server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitfinex/BitfinexExchange.java:704]: Should be warning

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitfinex/BitfinexExchange.java [hunk 7]
```diff
@@ -740,10 +745,8 @@ public int compare(LimitOrder lhs, LimitOrder rhs) {
                 log.debug("Called Bitfinex exchange for SELL rate: " + cryptoCurrency + fiatCurrency + " = " + tradableLimit);
                 return tradableLimit.multiply(cryptoAmount);
             }
-        } catch (ExchangeException e) {
-            log.error("Error", e);
-        } catch (IOException e) {
-            log.error("Error", e);
+        } catch (CurrencyPairNotValidException e) {
+            log.error("Currency pair not valid: {}, {}", currencyPair, e.getMessage());
```
Comment (generalbytes) [inline server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitfinex/BitfinexExchange.java:749]: Should be warning

**Review by generalbytes — CHANGES_REQUESTED** [review]

**Comment by pvyhnal-generalbytes** [comment]
changed all to warnings

**Pull request merged**